kompose/src/ui.tsx

```diff
@@ -0,0 +1,290 @@
+          console.error('PluginError:Kompose: fetching pods', err);
+          setProgressStatus(`Error: ${err.message}`);
+          setError(true);
+        },
+        {
+          queryParams: queryData,
+        }
+      )();
+    }
+
+    if (composeYAMLToConvert) {
+      runKompose();
+    }
+    return () => {
+      cancelled = true;
+
+      if (!!cancel) {
+        cancel.then(c => c());
+      }
+    };
+  }, [composeYAMLToConvert]);
+
+  const stepsRenderer = React.useCallback(() => {
+    if (step === 0) {
+      return (
+        <Box>
+          <MonacoEditor
+            value={composeYAML}
+            onChange={contents => {
+              setComposeYAML(contents);
+            }}
+            language={'yaml'}
+            height="500px"
+            options={{
+              selectOnLineNumbers: true,
+            }}
+            theme={theme.palette.mode === 'dark' ? 'vs-dark' : 'light'}
+          />
+        </Box>
+      );
+    }
+
+    if (step === 1) {
+      return (
+        <Box
+          sx={{
+            display: 'flex',
+            flexDirection: 'column',
+            alignItems: 'center',
+            justifyContent: 'center',
+            padding: 2,
+          }}
+        >
+          {!error ? (
+            <>
+              <Loader />
+              <Button
+                onClick={() => {
+                  setComposeYAMLToConvert('');
+                  setError(false);
+                  setStep(0);
+                }}
+                startIcon={<Icon icon="mdi:stop" width={20} />}
+                size="small"
+                variant="outlined"
+              >
+                Stop
+              </Button>
+            </>
+          ) : (
+            <Button
+              onClick={() => {
+                // @todo: The retry is not working apparently
+                setProgressStatus('');
+                setComposeYAMLToConvert(composeYAML);
+                setError(false);
+              }}
+              startIcon={<Icon icon="mdi:refresh" width={20} />}
+              variant="outlined"
+              size="small"
+            >
+              Retry
+            </Button>
+          )}
+          <Typography variant="body1" color={error ? 'error' : undefined} sx={{ paddingTop: 2 }}>
+            {progressStatus}
+          </Typography>
+        </Box>
+      );
+    }
+
+    return (
+      <Box>
+        <MonacoEditor
+          value={komposeOutput}
+          language={'yaml'}
+          height="500px"
+          options={{
+            selectOnLineNumbers: true,
+          }}
+          theme={theme.palette.mode === 'dark' ? 'vs-dark' : 'light'}
+        />
+      </Box>
+    );
+  }, [theme, komposeOutput, step, progressStatus, error]);
+
+  return (
+    <SectionBox title="Kompose (beta)" paddingTop={2}>
+      <Box>
+        <Typography variant="body1">
+          Kompose is a tool to help users who are familiar with docker-compose move to Kubernetes.
```

Comment:
Please mention that docker-compose files which contain `build` wont be converted, only compose files with container  image is accepted.